File: TangramJava/src/poo4/Term.java
package poo4;

import processing.core.PApplet;

class Term extends Shape {
  protected String _term;

  public Term(PApplet parent) {
    this(parent, 3, 8);
  }

  public Term(PApplet parent, String term, float scaling) {
    super(parent);
    setTerm(term);
    setScaling(scaling);
  }

  public Term(PApplet parent, int term, float scaling) {
    super(parent);
    setTerm(term);
    setScaling(scaling);
  }

  @Override
  protected void aspect() {
    _parent.noStroke();
    _parent.text(_term, 0, 0);
  }

  public String term() {
    return _term;
  }

  public void setTerm(String term) {
    _term = term;
  }

  public void setTerm(int term) {
    _term = new String();
    // see: https://discourse.processing.org/t/get-random-letters-and-put-into-a-string/28585/10
    for (int i = 0; i < term; i++) {
      _term += (char)((int)_parent.random(65, 65+24));
    }
  }
}

File: docs/js/term.js
class Term extends Shape {
    constructor (elements = 3, scaling = 8) {
      super();
      this.elements = elements;
      this.scaling = scaling;
    }
  
    aspect() {
      noStroke();
      text(this.elements, 0, 0);
    }
  
    get elements () {
      return this._term;
    }
  
    set elements (term) {
      // see: https://stackoverflow.com/questions/1349404/generate-random-string-characters-in-javascript
      this._term = typeof term === 'string' ? term : typeof term === 'number' ? Math.random().toString(36).slice(-term) : '👽';
    }
  }